.github/workflows/go-analyze.yml

```diff
@@ -0,0 +1,77 @@
+name: Go Analyze
+
+on:
```

Comment:
Can we add the paths to ignore like the original file?

Reply:
I moved to this approach so we can keep this a required check, but make it a noop for docs only updates. from my testing, there isn't an easy way to have required checks for only some code paths otherwise

Reply:
Ok so based on our discussion we want to have this job run every time and let the filter below to ignore the paths